src/modules/Blocks/store/sagas.ts

```diff
@@ -49,14 +51,33 @@ export function* getBlockList(action: ReturnType<typeof actions.getBlockList>) {
   }
 }
 
-function* watchGetBlockList() {
-  yield takeLatest(types.GET_BLOCK_LIST, getBlockList)
+function* blockPolling(action: ReturnType<typeof actions.getBlockList>) {
+  while (true) {
```

Comment:
while(true)  is a little error-prone, and no need to import a concept of  page(0) .
all you need is to add a judgement while get response from getBlockList , if current page is 1 then add a setTimeout.